packages/ember-application/tests/system/dependency_injection/to_string_test.js

```diff
@@ -1,62 +1,63 @@
 import { guidFor } from 'ember-utils';
-import { ENV, context } from 'ember-environment'; // lookup, etc
-import { run } from 'ember-metal';
-import Application from '../../../system/application';
+import { context } from 'ember-environment'; // lookup, etc
 import { Object as EmberObject } from 'ember-runtime';
-import DefaultResolver from '../../../system/resolver';
+
+import { moduleFor, AutobootApplicationTestCase, ModuleBasedTestResolver } from 'internal-test-helpers';
 
 let originalLookup, App;
 
-QUnit.module('Ember.Application Dependency Injection – toString', {
-  setup() {
+moduleFor('Ember.Application Dependency Injection - toString', class extends AutobootApplicationTestCase {
+  constructor() {
+    super();
     originalLookup = context.lookup;
 
-    run(() => {
-      App = Application.create();
-      context.lookup = {
-        App: App
-      };
+    this.runTask(() => {
+      App =  this.createApplication();
     });
-
+    context.lookup = {
+      App: App
+    }
     App.Post = EmberObject.extend();
-  },
+    this.add('model:post', App.Post);
+  }
 
   teardown() {
+    super.teardown();
     context.lookup = originalLookup;
-    run(App, 'destroy');
   }
-});
-
-QUnit.test('factories', function() {
-  let PostFactory;
-  PostFactory = App.__container__.factoryFor('model:post').class;
-  equal(PostFactory.toString(), 'App.Post', 'expecting the model to be post');
-});
-
-QUnit.test('instances', function() {
-  let post = App.__container__.lookup('model:post');
-  let guid = guidFor(post);
-
-  equal(post.toString(), '<App.Post:' + guid + '>', 'expecting the model to be post');
-});
-
-QUnit.test('with a custom resolver', function() {
-  run(App, 'destroy');
-
-  run(() => {
-    App = Application.create({
-      Resolver: DefaultResolver.extend({
-        makeToString(factory, fullName) {
-          return fullName;
-        }
-      })
-    });
-  });
 
-  App.register('model:peter', EmberObject.extend());
+  ['@test factories'](assert) {
```

Comment:
Above I put this test into a module with `DefaultResolver` in the module name. That's important, since this test is really testing `toString` on the default resolver.